spec/system/partner_system_spec.rb

```diff
@@ -196,4 +196,56 @@
       expect(partner.send_reminders).to be false
     end
   end
+
+  describe "#approve_partner" do
+    let(:tooltip_message) do
+      "Partner has not requested approval yet. \
+Partners are able to request approval going into 'My Organization' \
+and clicking 'Request Approval' button"
+    end
+    let!(:invited_partner) { create(:partner, name: "Amelia Ebonheart", status: :invited) }
+    let!(:awaiting_review_partner) { create(:partner, name: "Beau Brummel", status: :awaiting_review) }
+
+    context "when partner has :invited status" do
+      before { visit_approval_page(invited_partner.id, 1) }
+
+      it { expect(page).to have_selector(:link_or_button, 'Approve Partner') }
+      it { expect(page).to have_selector('span#pending-approval-request-tooltip > a.btn.btn-success.btn-md.disabled') }
+
+      it 'shows correct tooltip' do
+        page.execute_script('$("#pending-approval-request-tooltip").mouseover()')
+        expect(page).to have_content tooltip_message
+      end
+    end
+
+    context "when partner has :awaiting_review status" do
+      before { visit_approval_page(awaiting_review_partner.id, 2) }
+
+      it { expect(page).to have_selector(:link_or_button, 'Approve Partner') }
+      it { expect(page).not_to have_selector('span#pending-approval-request-tooltip > a.btn.btn-success.btn-md.disabled') }
+
+      it 'shows no tooltip' do
+        page.execute_script('$("#pending-approval-request-tooltip").mouseover()')
+        expect(page).not_to have_content tooltip_message
+      end
+    end
+  end
 end
+
+def stub_get_partner_request(partner_id)
+  allow(ENV).to receive(:[]).and_return(ENV)
+  allow(ENV).to receive(:[]).with("PARTNER_REGISTER_URL").and_return("https://partner-register.com")
+  allow(ENV).to receive(:[]).with("PARTNER_BASE_URL").and_return("https://partner-register.com")
+  allow(ENV).to receive(:[]).with("PARTNER_KEY").and_return("partner-key")
+
+  partner_response = File.read("spec/fixtures/partner_api/partner.json")
+
+  stub_request(:get, "https://partner-register.com/#{partner_id}")
+    .to_return(status: 200, body: partner_response, headers: {})
+end
```

Comment:
This might be deprecated very soon as the update to `main` to merge the two applications has been merged in. That means, we won't need to make this API call once we've updated the controller to directly retrieve from the partner's database rather than go through the API.

I think we can hold off on this until that change which should be coming soon